module/zstd/zstd.c

```diff
@@ -41,6 +41,19 @@
 MALLOC_DEFINE(M_ZSTD, "zstd", "ZSTD Compressor");
 #endif
 
+/*
+ * this specifies a incremental version of the zstd format
+ * increase it on any new added zstd version.
+ * this is made to ensure data integrity compatiblity since
+ * new versions might enhance the compression algorithm
+ * in a way the data itself will change.
+ */
+#define	ZSTD_VERSION		0
```

Comment:
what about using ZSTD_VERSION_NUMBER from zstdlib.h instead like @allanjude suggested?

Reply:
doesnt work. 16 bit limited. current zstd_version_number is already 10401. looks to fat for me. waste of numeric space.

Reply:
it's 0x28a5, which is 10405... and this IS 16 bit

Reply:
the max. version we could handle would be 6.55.35; that should be more than enough

Reply:
remember the year 2000, wait for year 2035. remember the 640kb limit. zfs is a very long term project already. no need for any risc here